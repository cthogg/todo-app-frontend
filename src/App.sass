@charset "utf-8";

// Import a Google Font
@import url('https://fonts.googleapis.com/css?family=Sniglet&display=swap');
// Set your brand colors
$orange: #FFE0B3;
$light-orange: #FFEDD3;
$dark-blue: #212A51
// Update Bulma's global variables
$section-padding: 0.5rem 1.5rem;
$family-sans-serif: 'Sniglet', cursive;
$text: $dark-blue;
$widescreen-enabled: false;
$fullhd-enabled: false;
$column-gap: 1rem
// Update some of Bulma's component variables
$body-background-color: $orange;
$card-background-color: $light-orange;
$control-border-width: 2px;
$input-border-color: transparent;
$input-shadow: none;

@import "../node_modules/bulma/bulma.sass";


.button
    border-color: $dark-blue
    background-color: $light-orange
/* TODO: add input background color */
.input


