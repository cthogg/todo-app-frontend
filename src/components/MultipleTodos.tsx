import React, { useState } from "react";
import { Todo } from "../types";
import SingleTodo from "./SingleTodo";

interface MultipleTodosProps {
  todos: Todo[];
  onDelete: Function;
}
const MultipleTodos: React.FunctionComponent<MultipleTodosProps> = ({
  todos,
  onDelete
}: MultipleTodosProps): JSX.Element => {
  const [filterInput, setFilterInput] = useState<string>("");
  const filteredTodos = todos.filter(todo =>
    todo.title.toLowerCase().includes(filterInput.toLowerCase())
  );
  return (
    <section className="section">
      <div className="card">
        <form>
          <label>
            Filter by Title:
            <input
              type="text"
              value={filterInput}
              onChange={e => setFilterInput(e.target.value)}
            />
          </label>
        </form>
        {filteredTodos.map((todo, i) => (
          <SingleTodo onDelete={id => onDelete(id)} key={i} todo={todo}>
            {" "}
          </SingleTodo>
        ))}
      </div>
    </section>
  );
};

export default MultipleTodos;
