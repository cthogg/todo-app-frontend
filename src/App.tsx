import React from "react";
import "./App.sass";
import TodoWrapper from "./components/TodoWrapper";
import { useAuth0 } from "./react-auth0-spa";
import NavBar from "./components/NavBar";

const App: React.FC = () => {
  const { isAuthenticated, loginWithRedirect, logout } = useAuth0();
  //TODO: change domain and audience in auth_config
  return (
    <div>
      <section className="section">
        <NavBar
          isAuthenticated={isAuthenticated}
          loginWithRedirect={loginWithRedirect}
          logout={logout}
        >
          {" "}
        </NavBar>
        <p className="is-size-2"> TODOs </p>
      </section>
      {isAuthenticated && <TodoWrapper> </TodoWrapper>}
    </div>
  );
};

export default App;
