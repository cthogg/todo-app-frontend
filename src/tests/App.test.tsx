import React from "react";
import App, { mockTodos } from "../App";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import MultipleTodos from "../components/MultipleTodos";

it("renders correctly", () => {
  const tree = renderer.create(<App> </App>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders multiple questions correctly", () => {
  const tree = renderer
    .create(
      <MultipleTodos todos={mockTodos} onDelete={() => null}>
        {" "}
      </MultipleTodos>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Renders TODO", () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/TODOS/i);
  expect(linkElement).toBeInTheDocument();
});
