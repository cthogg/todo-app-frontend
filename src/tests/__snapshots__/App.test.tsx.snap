// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<div>
  <section
    className="section"
  >
    <button
      className="button"
      onClick={[Function]}
    >
       
      Login
       
    </button>
    <div>
      Loading...
    </div>
    <p
      className="is-size-2"
    >
       TODOs 
    </p>
  </section>
  <section
    className="section"
  >
    <div
      className="card"
    >
      <p>
         
        Add a Task
      </p>
      <form
        action="#"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <input
          className="input"
          name="title"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Title"
          type="text"
          value=""
        />
        <input
          className="input"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Description"
          type="text"
          value=""
        />
        <input
          className="input"
          name="dueDate"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Date"
          type="date"
          value=""
        />
        <br />
        <button
          className="button"
          disabled={false}
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </section>
  <section
    className="section"
  >
     
    <div
      className="card"
    >
      <p>
         All Tasks Done 
      </p>
    </div>
     
  </section>
</div>
`;

exports[`renders TodoForm correctly 1`] = `
Array [
  <p>
     
    Add TODO
  </p>,
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <input
      className="input"
      name="title"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Title"
      type="text"
      value=""
    />
    <input
      className="input"
      name="description"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Description"
      type="text"
      value=""
    />
    <input
      className="input"
      name="dueDate"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Date"
      type="date"
      value=""
    />
    <br />
    <button
      className="button"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>,
]
`;

exports[`renders multiple questions correctly 1`] = `
Array [
  <section
    className="section"
  >
    <div
      className="card"
    >
      <form>
        <label>
          Filter by Title:
          <input
            className="input"
            onChange={[Function]}
            placeholder="Filter"
            type="text"
            value=""
          />
        </label>
      </form>
    </div>
  </section>,
  <section
    className="section"
  >
    <div
      className="card"
    >
      <div
        className="level"
      >
        <div
          className="level-left"
        >
          <span
            className="level-item button is-success"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
              />
            </svg>
             
          </span>
          <p
            className="level-item"
          >
             
            Friday, February 8th
             
          </p>
          <p
            className="level-item has-text-weight-bold"
          >
             
            Buy Milk
          </p>
          <p
            className="level-item"
          >
             
            Buy milk from Rewe
          </p>
        </div>
        <div
          className="level-right"
        >
          <button
            className="button"
            onClick={[Function]}
          >
             
            Edit
             
          </button>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="section"
  >
    <div
      className="card"
    >
      <div
        className="level"
      >
        <div
          className="level-left"
        >
          <span
            className="level-item button is-success"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
              />
            </svg>
             
          </span>
          <p
            className="level-item"
          >
             
            Friday, February 8th
             
          </p>
          <p
            className="level-item has-text-weight-bold"
          >
             
            Buy Pizza
          </p>
          <p
            className="level-item"
          >
             
            Ham and Pineapple
          </p>
        </div>
        <div
          className="level-right"
        >
          <button
            className="button"
            onClick={[Function]}
          >
             
            Edit
             
          </button>
        </div>
      </div>
    </div>
  </section>,
]
`;

exports[`renders single TODO correctly 1`] = `
<div
  className="level"
>
  <div
    className="level-left"
  >
    <span
      className="level-item button is-success"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
        />
      </svg>
       
    </span>
    <p
      className="level-item"
    >
       
      Friday, February 8th
       
    </p>
    <p
      className="level-item has-text-weight-bold"
    >
       
      Buy Milk
    </p>
    <p
      className="level-item"
    >
       
      Buy milk from Rewe
    </p>
  </div>
  <div
    className="level-right"
  >
    <button
      className="button"
      onClick={[Function]}
    >
       
      Edit
       
    </button>
  </div>
</div>
`;
